src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -186,18 +257,30 @@ export default class PhotoField extends React.Component {
         };
       }
     }
+
+    this.setState({ warningMessage: '' });
+  }
+
+  onZoomSliderChange = (e) => {
+    this.refs.cropper.zoomTo(e.target.value);
   }
 
   onZoom = (e) => {
     const zoomValue = e.detail.ratio;
     if (zoomValue < MAX_RATIO && zoomValue > MIN_RATIO) {
-      let warningMessage = null;
-      if (zoomValue >= WARN_RATIO) {
-        warningMessage = 'If you zoom in this close, your ID photo will be less clear.';
-      }
-      return this.setState({ zoomValue, warningMessage });
+      this.setState({ zoomValue });
+      this.updateBoundaryWarningAndButtonStates();
+
+      const moveBoundariesMet = getMoveBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
+      const zoomWarn = zoomValue >= WARN_RATIO;
+
+      this.setState({
+        warningMessage: makeCropBoundaryWarningMessage({ zoomWarn, ...moveBoundariesMet }),
+        ...makeMoveButtonsEnabledStates(moveBoundariesMet)
+      });
+    } else {
+      e.preventDefault();
```

Comment:
Do we need this?

Reply:
Looks like this prevents the image from being zoomed in by the buttons / mouse wheel beyond the boundaries.